src/pages/EventsPage.js

```diff
@@ -1,48 +1,61 @@
 import EventItem from '../components/EventItem.js';
 import EventInfo from '../components/EventInfo.js';
-import { useState, useContext } from 'react';
+import React, { useState, useContext } from 'react';
 import { EventsContext } from '../contexts/EventsContext.js';
-import { Link, Route, Switch, Redirect } from 'react-router-dom';
+import { Link, Route, Switch, Redirect, useRouteMatch } from 'react-router-dom';
 import ToDoList from '../components/ToDoList.js';
+import Header from '../components/Header.js';
+import MenuBar from '../components/MenuBar.js';
```

Comment:
Since the header and menuBar will be visible on all pages it would be good to only put them in App.js.

Reply:
I started with header and menu in App.js, but realized that we're accessing every page through events and we need to pass some data to these components (like a title for header) I thought this way would be much easier.

Reply:
Right, we access every page from the event page, I forgot about about that. So we only need to put the import here.
That way what I had in mind wouldn't be different from your solution, yours is simple and effective:D